src/hook/index.tsx

```diff
@@ -81,6 +79,10 @@ export const useRemixForm = <T extends FieldValues>({
     validKeys,
   );
 
+  const isValid = !(Object.keys(errors).length > 0);
+  const isSubmitted =
+    data && Object.keys(data).length > 0 && isValid ? true : false;
```

Comment:
you can just do `Object.keys(data??{}).length > 0 && isValid

Reply:
also when is `isSubmitted` set again? After the submission regardless if its valid or not?

Reply:
is submited is set as soon as the form is submitted. It will stay submitted until reset is called. So this should be isSubmittedSuccessfully instead.